//
// SmartCafe Project
// Copyright (C) 2016-2025 Dmitry Chapyshev <dmitry@aspia.ru>
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program. If not, see <https://www.gnu.org/licenses/>.
//

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package proto.port_forwarding;

message Config
{
    string remote_host  = 1;
    uint32 remote_port  = 2;
    uint32 local_port   = 3;
    string command_line = 4;
}

message Request
{
    string remote_host = 1;
    uint32 remote_port = 2;
}

message Result
{
    enum ErrorCode
    {
        UNKNOWN = 0;
        SUCCESS = 1;
        TIMEOUT = 2;
    }

    ErrorCode error_code = 1;
}

message Start
{
    uint32 dummy = 1;
}

message Data
{
    bytes data = 1;
}

message ClientToHost
{
    Request request = 1;
    Start start     = 2;
    Data data       = 3;
}

message HostToClient
{
    Result result = 1;
    Data data     = 2;
}
